utils/statistics-flow-reduce-operators.js

```diff
@@ -0,0 +1,83 @@
+'use strict';
+
+var isEmptyResult = function (processorRow) {
+	if (!processorRow) return true;
+	return !processorRow.approved && !processorRow.rejected &&
+		!processorRow.sentBack;
+};
+
+var buildResultByProcessor = function (row, certificate, processorId, date) {
+	var result = {
+		date: date,
+		processor: processorId,
+		processed: 0,
+		approved: 0,
+		sentBack: 0,
+		rejected: 0
+	};
+	['approved', 'sentBack', 'rejected'].forEach(function (status) {
+		if (!row[status]) return;
+		if (!result[status]) {
+			result[status] = 0;
+		}
+		if (!result.processed) {
+			result.processed = 0;
+		}
+		if (certificate) {
+			if (!row[status].certificate[certificate]) return;
+			result[status] += row[status].certificate[certificate];
+			result.processed += row[status].certificate[certificate];
+		} else {
+			result[status] += row[status].businessProcess;
+			result.processed += row[status].businessProcess;
+		}
+	});
+
+	return result;
+};
+
+module.exports = function (params) {
```

Comment:
It would be cleaner if `data`  and `query` are taken as separate arguments (`data` is not part of a `query`), it will also make invocation of that util more natural